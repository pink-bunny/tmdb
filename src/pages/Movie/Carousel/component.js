import React from 'react';
import { Row, Col, Carousel as AntdCarousel } from 'antd';
import PropTypes from 'prop-types';

const Carousel = ({ backdrops }) => (
  <Row type="flex">
    <Col span={24}>
      <AntdCarousel>
        {backdrops.map((item) => (
          <div key={item.file_path}>
            <img
              className="movie-image"
              src={`https://image.tmdb.org/t/p/original/${item.file_path}`}
              alt=""
            />
          </div>
        ))}
      </AntdCarousel>
    </Col>
  </Row>
);

Carousel.propTypes = {
  backdrops: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default Carousel;
