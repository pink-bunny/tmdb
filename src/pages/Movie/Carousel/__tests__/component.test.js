import React from 'react';
import { shallow } from 'enzyme';

import Carousel from '../component';

describe('Carousel component', () => {
  const requiredProps = {
    backdrops: [{
      /* eslint-disable camelcase */
      file_path: 'mock_path'
      /* eslint-enable camelcase */
    }]
  };

  it('matches snapshot with required props', () => {
    /* eslint-disable react/jsx-props-no-spreading */
    const wrapper = shallow(<Carousel {...requiredProps} />);
    /* eslint-enable react/jsx-props-no-spreading */
    expect(wrapper).toMatchSnapshot();
  });
});
