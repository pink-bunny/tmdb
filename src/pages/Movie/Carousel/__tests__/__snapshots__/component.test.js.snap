// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component matches snapshot with required props 1`] = `
<Row
  gutter={0}
  type="flex"
>
  <Col
    span={24}
  >
    <Carousel
      arrows={false}
      dots={true}
      draggable={false}
    >
      <div
        key="mock_path"
      >
        <img
          alt=""
          className="movie-image"
          src="https://image.tmdb.org/t/p/original/mock_path"
        />
      </div>
    </Carousel>
  </Col>
</Row>
`;
